src/main/java/org/spongepowered/common/mixin/core/command/MixinCommandExecuteAtSender.java

```diff
@@ -35,106 +39,132 @@
 import org.spongepowered.api.util.Tristate;
 import org.spongepowered.api.util.annotation.NonnullByDefault;
 import org.spongepowered.api.util.command.CommandSource;
+import org.spongepowered.api.util.command.source.ProxySource;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.common.command.WrapperCommandSource;
+import org.spongepowered.common.interfaces.IMixinCommandSender;
+import org.spongepowered.common.interfaces.IMixinCommandSource;
 
 import java.util.List;
 import java.util.Set;
 
-@Mixin(targets = "net/minecraft/command/CommandExecuteAt$1")
+@Mixin(targets = IMixinCommandSender.EXECUTE_COMMAND_SENDER)
 @NonnullByDefault
-public abstract class MixinCommandExecuteAtSender implements CommandSource, ICommandSender {
-    @Shadow(aliases = "val$sender")
-    private ICommandSender field_174802_b;
+public abstract class MixinCommandExecuteAtSender implements ProxySource, IMixinCommandSource, IMixinCommandSender {
+
+    @Shadow(aliases = "val$entity") private net.minecraft.entity.Entity field_174804_a;
+    @Shadow(aliases = "val$sender") private ICommandSender field_174802_b;
+    private MessageSink sink = Sponge.getGame().getServer().getBroadcastSink();
```

Comment:
seems like it makes more sense to get this from the source -- why have a separate one here? (or not at least default to the source's current sink)?